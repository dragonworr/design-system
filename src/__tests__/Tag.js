import React from 'react'
import renderer from 'react-test-renderer'
import { Tag, theme } from '..'

describe('Tag', () => {
  test('renders', () => {
    const json = renderer.create(<Tag />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('color gray sets color', () => {
    const json = renderer.create(<Tag color="gray" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.gray)
  })

  test('color blue sets color', () => {
    const json = renderer.create(<Tag color="blue" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.blue)
  })

  test('color green sets color', () => {
    const json = renderer.create(<Tag color="green" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.green)
  })

  test('color red sets color', () => {
    const json = renderer.create(<Tag color="red" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.red)
  })

  test('color orange sets color', () => {
    const json = renderer.create(<Tag color="orange" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.orange)
  })

  test('color purple sets color', () => {
    const json = renderer.create(<Tag color="purple" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.purple)
  })
})
