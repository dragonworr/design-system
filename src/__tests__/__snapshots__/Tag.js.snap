// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag color blue sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #007aff;
}

<div
  className="c0"
  color="blue"
/>
`;

exports[`Tag color gray sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #687B8E;
}

<div
  className="c0"
  color="gray"
/>
`;

exports[`Tag color green sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #0a0;
}

<div
  className="c0"
  color="green"
/>
`;

exports[`Tag color orange sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #fa0;
}

<div
  className="c0"
  color="orange"
/>
`;

exports[`Tag color purple sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #70b;
}

<div
  className="c0"
  color="purple"
/>
`;

exports[`Tag color red sets color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #f6f8fa;
  color: #c00;
}

<div
  className="c0"
  color="red"
/>
`;

exports[`Tag renders 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #d1d6db;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

<div
  className="c0"
  color="text"
/>
`;
